File: 1437.check-if-all-1-s-are-at-least-length-k-places-away.cs
/*
 * @lc app=leetcode id=1437 lang=csharp
 *
 * [1437] Check If All 1's Are at Least Length K Places Away
 */

// @lc code=start
public class Solution {
    public bool KLengthApart(int[] nums, int k) {
        int n = nums.Length;
        int prev = - 2*n;
        for (int i = 0; i < n; i++)
        {
            if (nums[i] == 1)
            {
                if (i - prev <= k)
                {
                    return false;
                }
                prev = i;
            }
        }
        return true;
    }
}
// @lc code=end

File: 1437.check-if-all-1-s-are-at-least-length-k-places-away.py
#
# @lc app=leetcode id=1437 lang=python3
#
# [1437] Check If All 1's Are at Least Length K Places Away
#

# @lc code=start
class Solution:
    # Minimum length must be between adjacent ones
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        prev = - float('Inf')
        for i, num in enumerate(nums):
            if num == 1:
                if i - prev <= k:
                    return False
                prev = i
        return True

# @lc code=end

